packages/aila/src/core/chat/AilaChat.ts

```diff
@@ -218,26 +227,80 @@ export class AilaChat implements AilaChatService {
     return applicableMessages;
   }
 
-  // #TODO this is the other part of the initial lesson state setting logic
-  // This should be some kind of hook that is specific to the
-  // generation of lessons rather than being applicable to all
-  // chats so that we can generate different types of document
+  // This method handles setting the initial state of the document
+  // based on the document type's implementation
   async handleSettingInitialState() {
     if (this._aila.document.hasInitialisedContentFromMessages) {
-      // #TODO sending these events in a different place to where they are set seems like a bad idea
-      const plan = this._aila.document.content;
-      const keys = Object.keys(plan) as Array<keyof typeof plan>;
-      for (const key of keys) {
-        const value = plan[key];
-        if (value) {
-          await this.enqueuePatch(`/${key}`, value);
+      const initialState = this._aila.document.getInitialState();
+
+      if (initialState) {
+        const keys = Object.keys(initialState);
+        for (const key of keys) {
+          const value = (initialState as Record<string, unknown>)[key];
+          if (value !== undefined && value !== null) {
+            const safeValue = this.ensureSafeValue(value);
+            if (safeValue !== undefined) {
+              await this.enqueuePatch(`/${key}`, safeValue);
+            }
+          }
         }
       }
     }
   }
 
+  /**
+   * Ensures values are safe to use in enqueuePatch using Zod validation
+   * with a maximum recursion depth to prevent stack overflow
+   */
+  private ensureSafeValue(
+    value: unknown,
+    depth: number = 0,
+  ): string | string[] | number | object | undefined {
+    if (depth > 10) {
+      return undefined;
+    }
+
+    try {
+      return this.safeValueSchema.parse(value);
+    } catch {
+      if (Array.isArray(value)) {
+        if (value.every((item) => typeof item === "string")) {
+          return value;
+        }
+
+        const safeArray = value
+          .map((item) => this.ensureSafeValue(item, depth + 1))
+          .filter(
+            (item): item is NonNullable<typeof item> => item !== undefined,
+          );
+
+        return safeArray.length > 0 ? safeArray : undefined;
+      }
+
+      if (value && typeof value === "object" && !Array.isArray(value)) {
+        const result: Record<string, unknown> = {};
+        let hasValidProperties = false;
+
+        for (const [key, propValue] of Object.entries(value)) {
+          const safeValue = this.ensureSafeValue(propValue, depth + 1);
+          if (safeValue !== undefined) {
+            result[key] = safeValue;
+            hasValidProperties = true;
+          }
+        }
+
+        return hasValidProperties ? result : undefined;
```

Comment:
I think you can drop hasValidProperties by using `Object.keys(result).length ? result : undefined`

That also opens up more functional options to build the object like a map and filter - though that's a code style thing